### ISSUE TITLE
python 3 doesn't detect correct number of installed nodes

### ISSUE THREAD

**Issue by Aaronsss** [RotorHazard/RotorHazard#497]
I have upgraded my installation to run python 3. The server no longer detects the correct number of nodes installed on the PCB.

I have a 6 node timer and the web interface shows 8 nodes available when Rotorhazard is run by python 3, the issue goes away when switching back to python 2

I am running the RotorHazard version 2.3.1

**Comment by HazardCreative** [comment]
What does your log look like?

**Comment by Aaronsss** [comment]
Having looked at it it seems like python 3 is using 'Using mock hardware interface' am I just being a noob?

Logs below from both versions

**Python 2**
RotorHazard v2.3.1
Release: 2.3.1 / Server API: 29 / Latest Node API: 25
Host machine: Raspberry Pi 3 Model B Rev 1.2
Host OS: Linux 5.4.83-v7+
Host machine is 'LapTimer' at 192.168.1.120
Using configuration file 'config.json'
Using log file: logs/rh_20210207_001402.log
Loaded module i2c_helper
Loaded module i2c_node
Loaded module serial_node
Searching for I2C nodes...
...I2C node 1 found at address 8
...I2C node 2 found at address 10
...I2C node 3 found at address 12
...I2C node 4 found at address 14
...I2C node 5 found at address 16
...I2C node 6 found at address 18
...No I2C node at address 20
...No I2C node at address 22
Node 1: API_level=25, Freq=1111, EnterAt=85, ExitAt=70
Node 2: API_level=25, Freq=5695, EnterAt=101, ExitAt=87
Node 3: API_level=25, Freq=1111, EnterAt=119, ExitAt=109
Node 4: API_level=25, Freq=1111, EnterAt=124, ExitAt=121
Node 5: API_level=25, Freq=1111, EnterAt=104, ExitAt=94
Node 6: API_level=25, Freq=1111, EnterAt=116, ExitAt=105
Number of nodes found: 6
Loaded module linux_sensor
Core temperature available
Set Profile to '3'
LED: selecting library "neopixel" (older)
LED: hardware GPIO enabled, count=150, pin=10, freqHz=800000, dma=10, invert=False, chan=0, strip=GRB/528384
Loaded module led_handler_strip
Running http server at port 5000
Interface: Starting background thread

**Python 3**
RotorHazard v2.3.1
Release: 2.3.1 / Server API: 29 / Latest Node API: 25
Host machine: Raspberry Pi 3 Model B Rev 1.2
Host OS: Linux 5.4.83-v7+
Host machine is 'LapTimer' at 192.168.1.120
Using configuration file 'config.json'
Using log file: logs/rh_20210207_001311.log
Loaded module i2c_node
Loaded module serial_node
Using mock hardware interface
Number of nodes found: 8
Loaded module linux_sensor
Core temperature available
Set Profile to '3'
LED: disabled (no modules available)
Running http server at port 5000
Interface: Starting background thread.

**Comment by ethomas997** [comment]
Have you done the pip-requirements install with the 'pip3' command, like this?:
```
cd ~/RotorHazard/src/server
sudo pip3 install -r requirements.txt
```

**Comment by Aaronsss** [comment]
Yep, without it the server wouldn't boot.

I tested with python 3.7.3 (I think). Am upgrading to python 3.9.1 to see if that helps

**Comment by ethomas997** [comment]
There are some code updates that may help with this.  If you'd like to try them out, please upgrade to the current "2.3" branch using the commands below and try it out.  (Before updating, any currently-running RotorHazard server should be stopped).
```
cd ~
wget https://codeload.github.com/RotorHazard/RotorHazard/zip/2.3 -O temp.zip
unzip temp.zip
mv RotorHazard RotorHazard.old
mv RotorHazard-2.3 RotorHazard
rm temp.zip
cp RotorHazard.old/src/server/config.json RotorHazard/src/server/
cp RotorHazard.old/src/server/database.db RotorHazard/src/server/

cd ~/RotorHazard/src/server
sudo pip install --upgrade --no-cache-dir -r requirements.txt
```
This will update the RH server version to 2.3.2.

**Comment by Aaronsss** [comment]
Thanks @ethomas997, using the above seems to have resolved the issue. I ran it on python 3.7.3

**Comment by ethomas997** [comment]
Ah great; thanks for testing and reporting.  We'll be posting a 2.3.2 version with these updates.

**Comment by HazardCreative** [comment]
Closed by f0c571d .

**Comment by frizzle101101** [comment]
i am quite the current instructions on the release page are incorrect because just attempted to set up a pi with those instructions and came here to find the solution please can you update the 2.3.2 release page instructions
